### DISCUSSION TITLE
TabularPredictor.refit_full:  How to combine sample weights and `refit_full`

### DISCUSSION THREAD

**Discussion by dluks** [autogluon/autogluon#4358] [Q&A] [answered]
## Description
I am using sample weights during bagged `TabularPredictor` fitting. I also require feature importance, so I leave out a validation set to compute that after initial fitting. For final model training I would like to include the held-out data with
```python
TabularPredictor.refit_full("best", train_data_extra=validation_set)
```
but this throws an error:

```console
Error training model: Applying 'sample_weight' with train_data_extra is not supported.
```

In this case, is it simply a no-go to include sample weights with `refit_full`? If not, is there no way to accommodate sample weights, a held-out validation set, and fitting models on the full data?

**Comment by Innixma** [comment, marked as answer]
Good question! This should be possible, it just hasn't been implemented yet. I created a GitHub issue to track: #4381. Please use the GitHub issue for further follow-up so we don't miss any messages, thanks!
